src/test/java/org/minions/devfund/sergio/SquarelotronTest.java

```diff
@@ -0,0 +1,347 @@
+    }
+
+    /**
+     * Tests the values in the square of 6x6.
+     */
+    @Test
+    public void testRotateWestIn6x6FourTurns() {
+        final int size = 6;
+        final int turns = -4;
+        final int[][] expected = {{1, 2, 3, 4, 5, 6}, {7, 8, 9, 10, 11, 12}, {13, 14, 15, 16, 17, 18},
+                {19, 20, 21, 22, 23, 24}, {25, 26, 27, 28, 29, 30}, {31, 32, 33, 34, 35, 36}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.rotateWest(turns)));
+    }
+
+    /**
+     * Tests the values of the method RotateRight.
+     */
+    @Test
+    public void testRotateRightWithPositiveTurns() {
+        final int size = 4;
+        final int[][] expected = {{16, 15, 14, 13}, {12, 11, 10, 9}, {8, 7, 6, 5},
+                {4, 3, 2, 1}};
+        squarelotron = new Squarelotron(size);
+        squarelotron.rotateRight(2);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values of the method RotateRight.
+     */
+    @Test
+    public void testRotateRightWithNegativeTurns() {
+        final int size = 4;
+        final int turns = -2;
+        final int[][] expected = {{16, 15, 14, 13}, {12, 11, 10, 9}, {8, 7, 6, 5},
+                {4, 3, 2, 1}};
+        squarelotron = new Squarelotron(size);
+        squarelotron.rotateRight(turns);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 3x3.
+     */
+    @Test
+    public void testUpsideDownFlipDimensionThreeRingOne() {
+        final int size = 3;
+        final int ring = 1;
+        final int[][] expected = {{7, 8, 9}, {4, 5, 6}, {1, 2, 3}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.upsideDownFlip(ring)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 3x3.
+     */
+    @Test
+    public void testUpsideDownFlipDimensionThreeRingTwo() {
+        final int size = 3;
+        final int ring = 2;
+        final int[][] expected = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.upsideDownFlip(ring)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 5x5.
+     */
+    @Test
+    public void testMainDiagonalFlipIn5x5RingOne() {
+        final int size = 5;
+        final int ring = 1;
+        final int[][] expected = {{1, 6, 11, 16, 21}, {2, 7, 8, 9, 22}, {3, 12, 13, 14, 23},
+                {4, 17, 18, 19, 24}, {5, 10, 15, 20, 25}};
+        squarelotron = new Squarelotron(size);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)